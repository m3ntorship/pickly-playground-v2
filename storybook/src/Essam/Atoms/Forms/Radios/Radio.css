/* Radios */
.radio-button {
  @apply relative;
}

.radio-button .radio-input {
  @apply absolute top-0 left-0 opacity-0;
  z-index: -1;
}

.radio-button .radio-btn {
  @apply w-6 h-6;
}

.radio-button.small .radio-btn {
  @apply w-4 h-4;
}

.radio-button .radio-btn,
.radio-button.small .radio-btn {
  @apply rounded-full border border-grey-shd4 block hover:border-grey-shd1 flex items-center justify-center;
}

.radio-button .radio-btn .circle-mark {
  @apply w-2.5 h-2.5 hidden;
}

.radio-button.small .radio-btn .circle-mark {
  @apply w-1.5 h-1.5;
}

.radio-button.small .radio-btn .circle-mark,
.radio-button .radio-btn .circle-mark {
  @apply bg-white rounded-full;
}

.radio-button .radio-input:checked + .radio-btn {
  @apply bg-primary hover:bg-primary-hover;
}

.radio-button .radio-input:focus + .radio-btn {
  @apply border-grey-shd3 border-2;
}

.radio-button .radio-input:checked + .radio-btn .circle-mark,
.radio-button .radio-input[disabled] + .radio-btn .circle-mark {
  @apply block;
}

.radio-button .radio-input[disabled] + .radio-btn {
  @apply border-grey-shd6 hover:border-grey-shd6;
}

.radio-button .radio-input[disabled][checked] + .radio-btn {
  @apply bg-primary-shd7 hover:bg-primary-shd7;
}

/* Radios */
