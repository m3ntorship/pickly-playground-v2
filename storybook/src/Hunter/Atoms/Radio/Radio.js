import React, { useState } from "react";
import PropTypes from "prop-types";

const Radio = ({ size = "small", disabled = false }) => {
  const [checked, setChecked] = useState(false);

  const iconSize = size === "small" ? "w-4 h-4" : "w-6 h-6";

  const isDisabled = disabled === true ? "pointer-events-none opacity-25" : "";

  const isChecked =
    checked && size === "small"
      ? "border-primary border-4"
      : checked && size === "default"
      ? "border-primary border-8"
      : "border border-grey-shd4 hover:border-grey-shd1";

  return (
    <div className="relative">
      <input type="radio" className="w-0 h-0 opacity-0 absolute" />
      <div
        className={`cursor-pointer rounded-full bg-white
                ${iconSize}
                ${isDisabled}
                ${isChecked}     
                `}
        onClick={() => setChecked(!checked)}
      ></div>
    </div>
  );
};

Radio.propTypes = {
  size: PropTypes.oneOf(["default", "small"]),
  disabled: PropTypes.bool,
};

export default Radio;
