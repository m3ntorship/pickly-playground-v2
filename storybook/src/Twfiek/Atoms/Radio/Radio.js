import React, { useState } from "react";
import PropTypes from "prop-types";

const Radio = (props) => {
  const { size = "lg", disabled = false } = props;
  const [checked, setChecked] = useState(false);
  return (
    <div className="relative">
      <input type="radio" className="w-0 h-0 opacity-0 absolute" />
      <div
        className={`cursor-pointer rounded-full bg-white ${
          size === "md" ? "w-4 h-4" : "w-6 h-6"
        } ${
          checked
            ? `${size === "md" ? "border-4" : "border-8"} border-primary`
            : "border border-grey-shd4 hover:border-grey-shd1"
        } ${disabled ? "pointer-events-none opacity-25" : ""}`}
        onClick={() => setChecked(!checked)}
      ></div>
    </div>
  );
};

Radio.propTypes = {
  checked: PropTypes.func,
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["md", "lg"]),
};

export default Radio;
