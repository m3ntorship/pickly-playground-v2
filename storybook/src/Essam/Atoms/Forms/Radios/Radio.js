import React from "react";
import PropTypes from "prop-types";
import className from "classnames";
import "./Radio.css";

export const Radio = ({ id, radioSize, ...props }) => {
  const inputId = id.trim().split(" ").join("");

  const RadioClasses = className("radio-button", {
    small: radioSize === "small",
  });
  return (
    <div className={RadioClasses}>
      <input type="radio" id={inputId} className="radio-input" {...props} />
      <label htmlFor={inputId} className="radio-btn">
        <div className="circle-mark"></div>
      </label>
    </div>
  );
};

Radio.propTypes = {
  id: PropTypes.string,
  radioSize: PropTypes.oneOf(["", "small"]),
};
Radio.defaultProps = {
  id: "",
  radioSize: "",
};
