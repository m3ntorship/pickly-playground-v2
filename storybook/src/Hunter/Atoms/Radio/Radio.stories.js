import React from "react";
import Radio from "./Radio";

export default {
  title: "Hunter/Atom/Radio",
  component: Radio,
  argTypes: {
    size: {
      control: {
        type: "radio",
        options: ["default", "small"],
      },
    },
  },
};

const Template = (args) => <Radio {...args} />;

export const radio = Template.bind({});
radio.args = {
  size: "default",
  disabled: false,
};
