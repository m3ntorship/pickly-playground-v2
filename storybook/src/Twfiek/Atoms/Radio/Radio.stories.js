import React from "react";
import Radio from "./Radio";

const defaultExport = {
  title: "Twfiek/Atoms/Radio",
  component: Radio,
};

export default defaultExport;

const Template = (args) => <Radio {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  disabled: false,
};
