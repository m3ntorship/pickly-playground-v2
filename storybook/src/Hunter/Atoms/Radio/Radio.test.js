import React from "react";
import renderer from "react-test-renderer";
import Radio from "./Radio";

describe("Radio component", () => {
  it("should render (default-unchecked) Radio when we apply nothing", () => {
    const tree = renderer.create(<Radio />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (disabled) Radio when we apply disabled", () => {
    const tree = renderer.create(<Radio disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (small) Radio when we apply small", () => {
    const tree = renderer.create(<Radio size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // it ("should render (checked) Radio when we apply checked", ()=>{\
  //     const tree = renderer.create(<Radio  />).toJSON();
  //     expect(tree).toMatchSnapshot();
  // });
});
