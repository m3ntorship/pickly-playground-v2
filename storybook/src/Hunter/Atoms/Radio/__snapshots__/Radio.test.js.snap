// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component should render (default-unchecked) Radio when we apply nothing 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    type="radio"
  />
  <div
    className="cursor-pointer rounded-full bg-white
                w-4 h-4
                
                border border-grey-shd4 hover:border-grey-shd1     
                "
    onClick={[Function]}
  />
</div>
`;

exports[`Radio component should render (default-unchecked) checkedbox when we apply nothing 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    type="radio"
  />
  <div
    className="cursor-pointer rounded-full bg-white
                w-4 h-4
                
                border border-grey-shd4 hover:border-grey-shd1     
                "
    onClick={[Function]}
  />
</div>
`;

exports[`Radio component should render (disabled) Radio when we apply disabled 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    type="radio"
  />
  <div
    className="cursor-pointer rounded-full bg-white
                w-4 h-4
                pointer-events-none opacity-25
                border border-grey-shd4 hover:border-grey-shd1     
                "
    onClick={[Function]}
  />
</div>
`;

exports[`Radio component should render (disabled) checkedbox when we apply disabled 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    type="radio"
  />
  <div
    className="cursor-pointer rounded-full bg-white
                w-4 h-4
                pointer-events-none opacity-25
                border border-grey-shd4 hover:border-grey-shd1     
                "
    onClick={[Function]}
  />
</div>
`;

exports[`Radio component should render (small) Radio when we apply small 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    type="radio"
  />
  <div
    className="cursor-pointer rounded-full bg-white
                w-4 h-4
                
                border border-grey-shd4 hover:border-grey-shd1     
                "
    onClick={[Function]}
  />
</div>
`;

exports[`Radio component should render (small) checkedbox when we apply small 1`] = `
<div
  className="relative"
>
  <input
    className="w-0 h-0 opacity-0 absolute"
    type="radio"
  />
  <div
    className="cursor-pointer rounded-full bg-white
                w-4 h-4
                
                border border-grey-shd4 hover:border-grey-shd1     
                "
    onClick={[Function]}
  />
</div>
`;
